plugin/data_masking/src/udf/udf_gen_dictionary_load.cc

```diff
@@ -95,6 +100,25 @@ static std::string _gen_dictionary_load(const char *dictionary_path,
   std::string s_dictname(dictionary_name);
   mysql::plugins::tolower(s_dictname);
 
+  /* Extract directory part of the filename */
+  char directory[FN_REFLEN]={0};
+  size_t dir_len = 0;
+  dirname_part(directory, dictionary_path, &dir_len);
+
+  if (dir_len == 0) {
+    std::stringstream ss;
+    ss << "ERROR: File path is not valid";
+    DBUG_RETURN(ss.str());
+  }
+
+  /* Check if dictionary file is present in --secure_file_priv directory */
+  if (!is_secure_file_path(directory)) {
+    std::stringstream ss;
```

Comment:
Do we need this stringstream? This error message looks like a fixed string.

Reply:
Not strictly. Originally I had secure_file_priv printed. The cost is negligible IMO

Reply:
Still, a simple string would be shorter too. Otherwise I agree, this isn't a major issue.

Reply:
addressed